### DISCUSSION TITLE
Chainlink VRF v1 vs V2

### DISCUSSION THREAD

**Discussion by PatrickAlphaC** [smartcontractkit/full-blockchain-solidity-course-py#1389] [Announcements]
So, Chainlink VRFv2 has been released, but we still use Chainlink V1 in the video. 

To follow along with the video & the docs, you can view [Chainlink VRFv1 docs here.](https://docs.chain.link/vrf/v1/examples/get-a-random-number)

**Comment by cromewar** [comment]
V2 to is just crazy and the Subscription Management is amazing!!

**Reply by Tanmaypatil123** [reply]
yeah, V2 is great but it became complicated for implementation.

**Comment by emilrueh** [comment]
@PatrickAlphaC the VRFv1 documentation is now linking to a 404 page.
Wondering which contract to use now to follow the tutorial or if one should learn VRFv2 now instead. Thanks!

EDIT: found the v1 documentation but under a new link so perhaps it could be good to change it in this topic :)
https://docs.chain.link/vrf/v1/examples/get-a-random-number

**Reply by cromewar** [reply]
@emilrueh 
Thanks for the hint, I changed the link to the new one :)

**Reply by emilrueh** [reply]
Thanks! Please also change it on the original tutorial repo under Lesson 7 below the code as it wants to link us to the v1 docs but redirects to 404: https://github.com/smartcontractkit/full-blockchain-solidity-course-py#lesson-7-smartcontract-lottery

![404vrf1](https://user-images.githubusercontent.com/103951556/204458698-112590e8-3925-4d93-8acf-6546bf1af06b.PNG)

**Reply by cromewar** [reply]
Thanks for the hint, the changes were made.
